client/signup/wpcom-login-form/index.jsx

```diff
@@ -1,63 +1,46 @@
 import config from '@automattic/calypso-config';
-import debugFactory from 'debug';
-import { Component } from 'react';
-
-const debug = debugFactory( 'calypso:signup:wpcom-login' );
-
-export default class WpcomLoginForm extends Component {
-	form = null;
-
-	componentDidMount() {
-		debug( 'submit form' );
-		this.form.submit();
+import { useEffect, useRef } from 'react';
+
+function getFormAction( redirectTo ) {
+	const subdomainRegExp = /^https?:\/\/([a-z0-9]+)\.wordpress\.com(?:$|\/)/;
```

Comment:
Let's add `-` to allowed subdomain characters. Here's a prior-art regexp: https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom-proxy-request/src/index.js#L540

It also allows a leading and trailing dashes which is probably bad, but not so much.

Reply:
Fixed via https://github.com/Automattic/wp-calypso/pull/68099/commits/48235ea0fa10559abd4c859b7dea4980f6d91725